src/services/app-storage.ts

```diff
@@ -192,6 +192,63 @@ export class AppStorage {
 		});
 	}
 
+	/**
+	 * Delete the data for a player.
+	 *
+	 * @param {Player} player The player to delete.
+	 * @memberof AppStorage
+	 */
+	public deletePlayer(player: Player): void {
+
+		// Finds groups that deleted player is in and deletes the group
+		this.getGroups().then((groups: Group[]) => {
+
+			// Go through each group and see if the deleted player is in the group.
+			// If deleted player is in group, then delete group.
+			groups.forEach((group: Group) => {
+				group.playerIds.forEach((memberId) => {
+					if(player.storageId === memberId) {
+
+						// Delete this group because the player is being deleted
+						this.deleteGroup(group);
+					}
+				});
+			});
+
+			this.deleteObject<Player>(player, this.playersCollection);
+
+		}).catch((reason: any) => {
+			console.error(`There was an error loading the groups. Error ${reason}. Did not delete player.`);
+		});
+	}
+
+	/**
+	 * Delete the group when a player is deleted.
```

Comment:
This comment should probably not specify that it's when a player is deleted since this could be called by something else (something that just deletes groups for example).